### ISSUE TITLE
Bug: pfm_range_list & pfm_range_list_titles not respected between 20 - 45 entries

### ISSUE THREAD

**Issue by apizz** [erikberglund/ProfileCreator#99]
I was working on an iTunes manifest and there is a preference - `ratingSystemID` - which defines the country to be used for the movie & tv show rating system with an `integer`.

However, it appears that somewhere between 20 - 45 items, rather than displaying a dropdown menu with the `pfm_range_list_titles`, instead this reverts into an integer entry field.

This iTunes preference has over 150 countries, so the number of list items would need to be increased pretty drastically to support this preference key.

**Comment by erikberglund** [comment]
Interesting.

There is a feature where if range_list is NOT set, but min and max is set for an integer value, and the range is less than 40, it will be converted to a popUpButton instead of a regular input filed.

This seems reversed if it's the case, I will have to look it up, because you are both adding the range list and the range list titles? Are you also adding the min and max values for the integer?

**Comment by erikberglund** [comment]
One thing, if you have both range_list and range_list_titles, if the item count is not exactly the same in both, the titles will be ignored, but that should not affect this either.

Could you send me the buggy manifest on slack and I will take a look at it.

**Comment by apizz** [comment]
I was pretty sure I counted exactly the number to ensure there was a string for each integer value.  You'll find the manifest attached.  I tried adding a pfm_range_max and pfm_range_min, but it just added a range label adjacent to the integer entry field, rather than turning back into a dropdown menu.

[com.apple.iTunes.plist.zip](https://github.com/erikberglund/ProfileCreator/files/2705535/com.apple.iTunes.plist.zip)

**Comment by apizz** [comment]
This is also a problem for one subkey preference in the Google Chrome manifest updates I'm working on.

The `ExtensionSettings` key has a number of subkeys.  One of these, `blocked_permissions` is within a dictionary & has 62 defined options, which are documented here: https://developer.chrome.com/extensions/declare_permissions

Despite supplying a `pmf_range_list`, it's only showing up as a text field and not a drop-down list.

![Screen Shot 2019-03-29 at 9 01 49 AM](https://user-images.githubusercontent.com/11789931/55234370-65ed3480-5201-11e9-9bc6-b47ef3340807.png)

```
<dict>
	<key>pfm_description</key>
	<string>Maps to a list of strings indicating the blocked API permissions for the extension.</string>
	<key>pfm_description_reference</key>
	<string></string>
	<key>pfm_name</key>
	<string>blocked_permissions</string>
	<key>pfm_subkeys</key>
	<array>
		<dict>
			<key>pfm_documentation_url</key>
			<string>https://developer.chrome.com/extensions/declare_permissions</string>
			<key>pfm_range_list</key>
			<array>
				<string>activeTab</string>
				<string>alarms</string>
				<string>background</string>
				<string>bookmarks</string>
				<string>browsingData</string>
				<string>certificateProvider</string>
				<string>clipboardRead</string>
				<string>clipboardWrite</string>
				<string>contentSettings</string>
				<string>contextMenus</string>
				<string>cookies</string>
				<string>debugger</string>
				<string>declarativeContent</string>
				<string>declarativeNetRequest</string>
				<string>declarativeWebRequest</string>
				<string>desktopCapture</string>
				<string>displaySource</string>
				<string>dns</string>
				<string>documentScan</string>
				<string>downloads</string>
				<string>enterprise.deviceAttributes</string>
				<string>enterprise.hardwarePlatform</string>
				<string>enterprise.platformKeys</string>
				<string>experimental</string>
				<string>fileBrowserHandler</string>
				<string>fileSystemProvider</string>
				<string>fontSettings</string>
				<string>gcm</string>
				<string>geolocation</string>
				<string>history</string>
				<string>identity</string>
				<string>idle</string>
				<string>idltest</string>
				<string>management</string>
				<string>nativeMessaging</string>
				<string>networking.config</string>
				<string>notifications</string>
				<string>pageCapture</string>
				<string>platformKeys</string>
				<string>power</string>
				<string>printerProvider</string>
				<string>privacy</string>
				<string>processes</string>
				<string>proxy</string>
				<string>sessions</string>
				<string>signedInDevices</string>
				<string>storage</string>
				<string>system.cpu</string>
				<string>system.display</string>
				<string>system.memory</string>
				<string>system.storage</string>
				<string>tabCapture</string>
				<string>tabs</string>
				<string>topSites</string>
				<string>tts</string>
				<string>ttsEngine</string>
				<string>unlimitedStorage</string>
				<string>vpnProvider</string>
				<string>wallpaper</string>
				<string>webNavigation</string>
				<string>webRequest</string>
				<string>webRequestBlocking</string>
			</array>
			<key>pfm_type</key>
			<string>string</string>
		</dict>
	</array>
	<key>pfm_type</key>
	<string>array</string>
</dict>
```

**Comment by erikberglund** [comment]
This last thing is because the ui items inside that is probably not added. So I should add it in.

**Comment by erikberglund** [comment]
The other popupbutton issue I have found and solved now.

**Comment by apizz** [comment]
Confirmed this is working.  Only challenge is selecting the item, rather than the dropdown menu, to remove it.

[Chrome_popup_dropdown_menu_UI_exp.zip](https://github.com/erikberglund/ProfileCreator/files/3027103/Chrome_popup_dropdown_menu_UI_exp.zip)

**Comment by apizz** [comment]
Fixed in 0.2.5 release
